packages/babel-plugin/src/utils/css-builders.ts

```diff
@@ -656,13 +659,12 @@ const extractTemplateLiteral = (node: t.TemplateLiteral, meta: Metadata): CSSOut
       return acc + quasi.value.raw + interpolation.value;
     }
 
-    if (
+    const doesExpressionContainCssBlock =
       t.isObjectExpression(interpolation) ||
       isCompiledCSSTaggedTemplateExpression(interpolation, meta.state) ||
-      isCompiledCSSCallExpression(interpolation, meta.state) ||
-      (t.isArrowFunctionExpression(nodeExpression) &&
-        t.isConditionalExpression(nodeExpression.body))
-    ) {
+      isCompiledCSSCallExpression(interpolation, meta.state);
+
+    if ((!isMidStatement && doesExpressionContainCssBlock) || doesExpressionHaveConditionalCss) {
```

Comment:
Without the `isMidStatement` this was breaking on styles like:
`${({ color }) => (color && COLOR_MAP[color] ? COLOR_MAP[color] : colors.B400)};`
Because it `color` is dynamic, Compiled can only work out what `COLOR_MAP` is. It saw it was an `ObjectExpression` and spread the whole Object into the stylesheet.

We might want to consider returning the original node/expression if we can't evaluate it the whole way through for certain expression types.